service/src/main/java/bio/terra/workspace/service/resource/controlled/flight/clone/bucket/CreateStorageTransferServiceJobStep.java

```diff
@@ -63,45 +71,45 @@ public StepResult doStep(FlightContext flightContext)
     logger.info(
         "Creating transfer job named {} in project {}", transferJobName, controlPlaneProjectId);
 
+    // Look up the transfer job by name. If it's found, it means we are restarting this step and
+    // the job either has an operation in progress or completed (possibly failed).
     try {
-      final Storagetransfer storageTransferService =
-          StorageTransferServiceUtils.createStorageTransferService();
-
-      // Look up the transfer job by name. If it's found, it means we are restarting this step and
-      // the job either has an operation in progress or completed (possibly failed).
-      try {
-        final TransferJob existingTransferJob =
-            storageTransferService
-                .transferJobs()
-                .get(transferJobName, controlPlaneProjectId)
-                .execute();
-        if (null != existingTransferJob) {
-          logger.info(
-              "Transfer Job {} already exists. Nothing more for this step to do.", transferJobName);
-          return StepResult.getStepResultSuccess();
-        }
-      } catch (GoogleJsonResponseException e) {
-        logger.info("No pre-existing transfer job named {} found.", transferJobName);
+      final TransferJob existingTransferJob =
+          storagetransfer
+              .transferJobs()
+              .get(transferJobName, controlPlaneProjectId)
+              .execute();
+      if (null != existingTransferJob) {
+        logger.info(
+            "Transfer Job {} already exists. Nothing more for this step to do.", transferJobName);
+        return StepResult.getStepResultSuccess();
       }
-      // Get the service account in the control plane project used by the transfer service to
-      // perform the actual data transfer. It's named for and scoped to the project.
-      // Storage transfer service itself is free, so there should be no charges to the control
-      // plane project. The usual egress charges will be made on the source bucket.
-      // TODO(PF-888): understand what happens when the source bucket is requester pays. We don't
-      //   support requester pays right now for controlled gcs buckets, but it might be set on a
-      //   referenced bucket.
-      final String transferServiceSAEmail =
-          workingMap.get(ControlledResourceKeys.STORAGE_TRANSFER_SERVICE_SA_EMAIL, String.class);
-      logger.debug("Storage Transfer Service SA: {}", transferServiceSAEmail);
+    } catch (GoogleJsonResponseException e) {
+      logger.info("No pre-existing transfer job named {} found.", transferJobName);
+    } catch (IOException e) {
+      return new StepResult(
+          StepStatus.STEP_RESULT_FAILURE_FATAL,
+          new IllegalStateException("Failed to check for existing storage transfer job.", e));
```

Comment:
Any reason to retry the step here? Or for the other transfer service calls below?

Reply:
You mean reason not to retry it? I guess we can retry.

Reply:
Our general strategy has been to go fatal and then discover through integration testing what cases we should retry. It is not necessarily a bad strategy. What I want to know is, do we know of any error responses from the transfer service that we should retry? Or do we have to find and weed them later?